clang/include/clang/3C/RewriteUtils.h

```diff
@@ -87,20 +87,22 @@ class FunctionDeclReplacement
                                   DeclReplacement::DRK_FunctionDecl> {
 public:
   explicit FunctionDeclReplacement(FunctionDecl *D, std::string R, bool Return,
-                                   bool Params)
-      : DeclReplacementTempl(D, nullptr, R), RewriteReturn(Return),
-        RewriteParams(Params) {
+                                   bool Params, bool Generic = false)
+      : DeclReplacementTempl(D, nullptr, R), RewriteGeneric(Generic),
+        RewriteReturn(Return), RewriteParams(Params) {
     assert("Doesn't make sense to rewrite nothing!" &&
-           (RewriteReturn || RewriteParams));
+           (RewriteGeneric || RewriteReturn || RewriteParams));
   }
 
   SourceRange getSourceRange(SourceManager &SM) const override;
 
 private:
   // This determines if the full declaration or the return will be replaced.
+  bool RewriteGeneric;
```

Comment:
This is `true` when you need to add `_Itype_for_any(...)` I'm guessing? Not that if this function has a normal checked type, i.e., is not an itype, you should rewrite with `_For_any(...)` instead.

Reply:
I've never seen `_For_any(...)` used and didn't consider it. I'll have to see what I can do to integrate that into this change.